.github/workflows/bazel-lockfile.yml

```diff
@@ -0,0 +1,221 @@
+# Workflow to automatically update the bazel lockfile on pull requests and dev
+# branches. In general, fails if lockfile is out of date.
+# For each platform we support, we run a dry (--nobuild) Bazel build, and then
+# upload the resulting lockfile as an artifact. The next platform downloads the
+# lockfile artifact, and then also runs a dry build, rinse, repeat.
+name: Update MODULE.bazel.lock
+on:
+  # Merge groups should be checked for up-to-date lockfile
+  merge_group:
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+  push:
+    branches:
+      # master is checked for up-to-date lockfile
+      - master
+      # dev branches get a new commit with updated lockfile (when necessary)
+      - 'dev-gh-*'
```

Comment:
Shouldn't there be a `pull_request` trigger here?

Reply:
good catch :)